### ISSUE TITLE
Category management

### ISSUE THREAD

**Issue by naglepuff** [DigitalSlideArchive/wsi-superpixel-guided-labeling#69]
Use cases:

- Delete category + unlabel
- Merge category

**Comment by jagstein** [comment]
. it would be helpful to have a single source of truth (e.g. a .yaml file) that can be edited using the UI, hand edited, or moved between projects.
. Category management should be available in the Initial Labeling task
. Unclear whether category management is needed during the Active Learning / Improve Predictions task
. Likely need to build a dedicated UI/task for managing categories

What information do we need about each category?
. key -- internal value generated by HistomicsUI -- unchangeable. If users are adding a category by editing the .yaml, they should leave this blank and then HistomicsUI generates a key
. Full title
. Short title for display
. Edge color & thickness
. Fill color & opacity
. hotkey --> (if people are editing by hand unclear how we would prevent collisions)
. order (if people are editing by hand and the categories are given the same order, we can alphabetize)
. ??? anything else

**Comment by jagstein** [comment]
Categories still randomly switching in the initial labeling task
I labeled 100s of patches including many marker (see below) which are no longer being counted??

![Image](https://github.com/DigitalSlideArchive/wsi-superpixel-guided-labeling/assets/6061875/bbc38279-2564-47a2-9b97-b4698541e3d8)

**Comment by bnmajor** [comment]
@jagstein Is this happening with the latest changes on master? And if so are you seeing the config file that should have been created in the project directory?

**Comment by jagstein** [comment]
That was the old version. Haven't seen that with the new version. I have seen this:

Hotkey reverts when trying to set a new hotkey:



![Image](https://github.com/DigitalSlideArchive/wsi-superpixel-guided-labeling/assets/6061875/6a1ba4f9-28f4-40f1-91a9-92a75dc4afa3)



![Image](https://github.com/DigitalSlideArchive/wsi-superpixel-guided-labeling/assets/6061875/fd22291d-97d2-463f-a8f0-0fecf858af40)



![Image](https://github.com/DigitalSlideArchive/wsi-superpixel-guided-labeling/assets/6061875/5a0fe75f-834e-4aad-995f-73a4fa38308b)

**Comment by bnmajor** [comment]
@jagstein Thanks for the update!

To set the hotkeys you will need to:
- Press "enter" once you've set the new hot key you'd like to use. If the value is still in italics the change has not been committed.
- Press the check mark to accept all changes (hot key use will be prevented until you're done making changes)

If you are doing both of these steps and still seeing issues though please let me know!

**Comment by jagstein** [comment]
Squirrely .yamls:
. Project A is going well, has been trained and retrained a few times.
. Project B is new. Slides uploaded. 
. Project B shares some categories with project A. I copy over the .yaml and then delete the unneeded categories.
. Start initial labeling task in project B --> unneeded categories appear in the list.
. Look at .yaml for project B --> file has duplicated, categories are back
. Delete categories in .yaml. save.
. Exit yaml and recheck - categories are still gone.
. Go to labeling task for project B --> unneeded categories are back
. Look at .yaml for project B --> file has duplicates, categories are back

**Comment by bnmajor** [comment]
@jagstein I'm very sorry, I'm not sure why I missed this comment!

Before the latest labeling changes were merged in there was no way to delete categories, including through the config file so once a category existed it was there forever. Master now has the delete and merge features that I demoed at the last meeting so if you pull the latest changes again you will be able to remove these categories through the UI.

Even without the option to delete though I see a few other things that could be bugs:

> . Project B shares some categories with project A. I copy over the .yaml and then delete the unneeded categories.
. Start initial labeling task in project B --> unneeded categories appear in the list.

It sounds like you copied the yaml and deleted the unneeded categories *before* you opened the UI to start the initial labeling. In that case those extraneous categories should not have even been created. Is it possible that you opened the UI before you deleted the categories? And if not, did you open project B in the Histomics UI or some other way? I am trying to figure out where it may have picked up the extra categories from...

> . Look at .yaml for project B --> file has duplicated, categories are back
...
. Look at .yaml for project B --> file has duplicates, categories are back

There should definitely not be duplicates. To clarify - are you saying there is a duplicate file? Or that there are duplicate categories in the file?
